.github/workflows/lintcommit.js

```diff
@@ -0,0 +1,157 @@
+//
+// Checks that a PR title conforms to our custom flavor of "conventional commits"
+// (https://www.conventionalcommits.org/).
+//
+// To run self-tests, simply run this script:
+//
+//     node lintcommit.js test
+//
+// TODO: also check the subject line of the first commit message in the PR.
+// TODO: this script intentionally avoids github APIs so that it is locally-debuggable, but if those
+// are needed, use actions/github-script as described in: https://github.com/actions/github-script?tab=readme-ov-file#run-a-separate-file
+//
+
+const fs = require('fs')
```

Comment:
Can we re-use https://github.com/wagoid/commitlint-github-action as pre-commit or ci?

Reply:
I intentionally did not do that. The 3P actions for "conventional commits" are very over-engineered. It's much simpler to customize a single-file script instead of learning all the various "config options" for the 3P actions (assuming they have the configurability we want).